include/dali/c_api.h

```diff
@@ -205,13 +205,15 @@ daliSetExternalInputTensorsAsync(daliPipelineHandle *pipe_handle, const char *na
                                  device_type_t device, const void *const *data_ptr,
                                  dali_data_type_t data_type, const int64_t *shapes,
                                  int64_t sample_dim, const char *layout_str,
-                                 cudaStream_t stream, int sync, int is_pinned);
+                                 cudaStream_t stream, int sync, int is_pinned,
```

Comment:
I wonder if in case of repetetive arguments of the same time like here: 
`int sync, int is_pinned, int no_copy` that can start to be confusing at the call site the more of them we add, maybe we should fill a config structure and pass it (similarly to what kernel does).

Reply:
Good idea. But I want to wait for the second opinion as this will shape our C APi.

Reply:
IMO this calls for `unsigned flags`

```
enum {
  DALI_ext_default = 0,  // we can put whatever we like here
  DALI_ext_force_sync = (1<<0),
  DALI_ext_pinned = (1<<1),
  DALI_ext_no_copy =(1<<2)
};
```

Reply:
Done